### ISSUE TITLE
Not working with Phonegap Build?

### ISSUE THREAD

**Issue by Papasmurfo** [phonegap-build/StatusBarPlugin#26]
Is this not working with Phonegap build?

Currently trying to change my status bar colour, with the following code:

```
    <gap:plugin name="com.phonegap.plugin.statusbar" />
    <preference name="StatusBarBackgroundColor" value="#4e9fd5" />
```

Is there any specific order I should be putting these in or in certain places? 

Thanks for the plugin! Any help is greatly appreciated,

Jonathan

**Comment by wildabeast** [comment]
The `StatusBarBackgroundColor` and `StatusBarOverlaysWebView` preference tags are not yet supported by PhoneGap Build ([in the readme](https://github.com/phonegap-build/StatusBarPlugin#preferences)).

Try this instead:

```
function deviceReady() {
    StatusBar.backgroundColorByHexString("#4e9fd5");
}
document.addEventListener('deviceready', deviceReady, false);
```

**Comment by Papasmurfo** [comment]
Thanks will give it a try!

**Comment by Papasmurfo** [comment]
This code makes the background of the status bar 'black' with black text.

Is there any other code that could be a work around?

Thanks again for the help,

Jonathan

**Comment by igos** [comment]
When it will be supported by Phonegap build?

**Comment by wildabeast** [comment]
@Papasmurfo What are you trying to achieve? If you want black with white text, check [the docs](https://github.com/phonegap-build/StatusBarPlugin/blob/master/README.md):

```
StatusBar.backgroundColorByHexString("#4e9fd5");
StatusBar.styleLightContent();
```

**Comment by wildabeast** [comment]
@Igos not entirely sure -- its on the backlog but not a priority, since you can easily work around it by setting these values via javascript instead.

**Comment by Papasmurfo** [comment]
Hi Ryan,

I tried the code that you supplied but the background colour doesn't work,
but it does "kind" of work as it changes the background to black? With
black text?

Any help is greatly appreciated,

Jonathan

**Comment by wildabeast** [comment]
@Papasmurfo did you try:

```
StatusBar.styleLightContent();
```

**Comment by Papasmurfo** [comment]
Yes - copy and pasted the code you supplied before. I think it was wrapped
in a function.

Thanks again,

Jonathan

**Comment by wildabeast** [comment]
Is this on PhoneGap Build? Can you share your App ID?

**Comment by Papasmurfo** [comment]
Yes, would you be able to supply your email so I can throw you the whole JS
file and then we can post the resolution here.

Jonathan

**Comment by wildabeast** [comment]
Please send it to build at phonegap.com

**Comment by Papasmurfo** [comment]
Ah - I'll just post the code here but essentially in my main.js file I have
the following.

function deviceReady() {
    StatusBar.backgroundColorByHexString("#4e9fd5");
}
document.addEventListener('deviceready', deviceReady, false);

This seems to create a black bar with black text for some reason.

On Friday, March 28, 2014, Ryan Willoughby
<notifications@github.com<javascript:_e(%7B%7D,'cvml','notifications@github.com');>>
wrote:

**Comment by wildabeast** [comment]
I can't really do much with that without seeing the context. Please send me your PhoneGap Build app ID to build at phonegap.com.

**Comment by Papasmurfo** [comment]
Will do Ryan. Thanks again for helping our with this one.

Best,

Jonathan

**Comment by wildabeast** [comment]
@Papasmurfo haven't heard from you, closing this issue. Let me know if I missed an email.

**Comment by sfentress** [comment]
@wildabeast I can replicate @Papasmurfo's bug on Phonegap build. Here's my build: https://build.phonegap.com/apps/951372/builds

I am trying to call `StatusBar.backgroundColorByHexString("#FFFFFF");` but instead I get a black statusbar with black text.

As far as I can tell, `StatusBar` is never defined -- if I use Phonegap Build's debug view I get `ReferenceError: Can't find variable: StatusBar`

**Comment by sfentress** [comment]
I was able to fix this by switching to version 1.0.1 of the plugin. Using the earlier version, I can now access the StatusBar JS object, and set the background color to white.

**Comment by omnoms** [comment]
Yeah, I have the same issue. StatusBar is undefined in the global scope for javascript in the latest. Had to revert to 1.0.1
